modules/rhpam-apb/roles/generate-secret/tasks/main.yml

```diff
@@ -32,9 +32,66 @@
               service: '{{ service_name }}'
           data: "{{ lookup('template', 'secret.yml.j2') | from_yaml }}"
   when:
+    - not amq_secret
     - secret_cmd.rc != 0
     - state == 'present'
 
+- name: "[AMQ] Generate secret if it does not exist"
+  block:
+    - name: "[AMQ] Generate a self-signed certificate for the broker keystore"
+      shell: >
+        keytool -genkey -alias broker -keyalg RSA -keystore {{ keystore_path }} -dname "CN={{ console_host }}" -storepass {{ keystore_pwd }}
```

Comment:
use `-genkeypair` instead of `-genkey` see https://docs.oracle.com/javase/8/docs/technotes/tools/unix/keytool.html#keytool_option_genkeypair
use `{{ keystore_alias }}` instead of `broker`
use `{{ route_host }}` instead of `{{ console_host }}`